clio-client/src/main/scala/org/broadinstitute/clio/client/dispatch/DeliverArraysExecutor.scala

```diff
@@ -20,68 +19,60 @@ import scala.concurrent.ExecutionContext
   *   2. Records the updated locations of the idat files, along with the workspace name, in the metadata
   */
 class DeliverArraysExecutor(deliverCommand: DeliverArrays)(implicit ec: ExecutionContext)
-    extends MoveExecutor(deliverCommand) {
+    extends DeliverExecutor(deliverCommand) {
 
-  override protected[dispatch] def buildMove(
-    metadata: ArraysMetadata,
-    ioUtil: IoUtil
+  override protected[dispatch] def buildDelivery(
+    movedMetadata: ArraysMetadata,
+    moveOps: immutable.Seq[IoOp]
   ): Source[(ArraysMetadata, immutable.Seq[IoOp]), NotUsed] = {
+    (movedMetadata.grnIdatPath, movedMetadata.redIdatPath) match {
+      case (Some(grn), Some(red)) => {
+        val movedGrnIdat = Metadata.findNewPathForMove(
+          grn,
+          deliverCommand.destination,
+          ArraysExtensions.IdatExtension
+        )
+        val movedRedIdat = Metadata.findNewPathForMove(
+          red,
+          deliverCommand.destination,
+          ArraysExtensions.IdatExtension
+        )
+        val idatCopies = immutable
+          .Seq(
+            CopyOp(grn, movedGrnIdat),
+            CopyOp(red, movedRedIdat)
+          )
+          .filterNot { op =>
+            op.src.equals(op.dest)
+          }
 
-    val baseStream = super
-      .buildMove(metadata, ioUtil)
-      .orElse(Source.single(metadata -> immutable.Seq.empty))
-
-    baseStream.flatMapConcat {
-      case (movedMetadata, moveOps) =>
-        (movedMetadata.grnIdatPath, movedMetadata.redIdatPath) match {
-          case (Some(grn), Some(red)) => {
-            val movedGrnIdat = Metadata.findNewPathForMove(
-              grn,
-              deliverCommand.destination,
-              ArraysExtensions.IdatExtension
-            )
-            val movedRedIdat = Metadata.findNewPathForMove(
-              red,
-              deliverCommand.destination,
-              ArraysExtensions.IdatExtension
-            )
-            val idatCopies = immutable
-              .Seq(
-                CopyOp(grn, movedGrnIdat),
-                CopyOp(red, movedRedIdat)
-              )
-              .filterNot { op =>
-                op.src.equals(op.dest)
-              }
-
-            val newMetadata = movedMetadata
-              .withWorkspaceName(deliverCommand.workspaceName)
-              .copy(
-                grnIdatPath = Some(movedGrnIdat),
-                redIdatPath = Some(movedRedIdat)
-              )
+        val newMetadata = movedMetadata
+          .withWorkspaceName(deliverCommand.workspaceName)
```

Comment:
Could we move the call to `withWorkspaceName` into the superclass?

Reply:
OK